src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -46,6 +46,22 @@ internal sealed class VariableState
             }
         }
 
+        /// <summary>
+        /// Represents the result of visiting an expression.
+        /// Contains a result type which tells us whether the expression may be null,
+        /// and an l-value type which tells us whether we can assign null to the expression.
+        /// </summary>
+        private struct VisitResult
```

Comment:
>struct [](start = 16, length = 6)

`readonly struct`

Reply:
`ResultType` and `_lValueType` can be set.

---
In reply to: [258110524](https://github.com/dotnet/roslyn/pull/33447#discussion_r258110524) [](ancestors = 258110524)

Reply:
Can the struct be made immutable though?

---
In reply to: [258231060](https://github.com/dotnet/roslyn/pull/33447#discussion_r258231060) [](ancestors = 258231060,258110524)